### PR TITLE
Update Web3 Provider

### LINKED ISSUE #1: Bug: Attempting to connect via WalletConnect crashes webapp
- Up to date Chrome
- Clear all site data for https://govern.sarcophagus.io
- Go to site
- Click `Connect Web3 Account`
- Select "WalletConnect"
- Use wallet that supports WalletConnect to scan barcode (I was using Rainbow on iOS)
- As soon as app scans barcode, WalletConnect modal closes and webapp crashes soon after that

Examining the console during a crash shows many `wallet connect` log messages being output, before eventually idk running out of stack space or whatever.

Screen Shot 2022-06-17 at 3 32 24 PM

cc: @sethhrbek just so you're aware, i'm opening this issue

### PR DESCRIPTION
**Pull request by dacebt**
## Overview
closes #1 
Updates dApp with lastest code for Web3 Provider (A Javascript version). This version has been updated to work with the current ENVs. so no secrets will need to be changed

## Notes

Note that most the file changes are due to updating to new provider object.

## Other Updates
- went ahead and added some updates to the toastContainer
- updated provider toast to dark theme
- found and fix a bug that would happen when revoking access (mannually disconnecting)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "@testing-library/react": "^11.2.5",
     "@testing-library/user-event": "^12.8.3",
     "@tippyjs/react": "^4.2.4",
-    "@walletconnect/web3-provider": "^1.4.0",
+    "@walletconnect/ethereum-provider": "^1.7.8",
```
Comment (dacebt) [inline package.json:12]: This updates to updates to newest version of wallet connect

File: src/components/Body.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
 import React from 'react'
-import { useWeb3 } from '../web3'
-import { useAddresses } from '../web3/chains'
+import { useAddresses } from '../context/blockchain/useAddresses'
```
Comment (dacebt) [inline src/components/Body.js:2]: extracted from old `chains.js` to own hook in context directory

File: src/context/web3Data/chains.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const supportedChains = () => {
+  const dev = process.env.NODE_ENV !== 'production' ? [parseInt(process.env.REACT_APP_CHAIN_ID, 10)] : []
+  return [...dev, parseInt(process.env.REACT_APP_CHAIN_ID, 10)]
+}
```
Comment (dacebt) [inline src/context/web3Data/chains.js:4]: this uses the old `chains.js` code

File: src/context/web3Data/hooks/useLIsteners.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    // unsubscribe
+    return () => {
+      if(modalProvider.isWalletConnect) {
+        modalProvider.off("accountsChanged");
+        modalProvider.off("chainChanged");
+        modalProvider.off("disconnect");
+      } else {
+        modalProvider.removeAllListeners();
+      }
```
Comment (dacebt) [inline src/context/web3Data/hooks/useLIsteners.js:81]: This section of code will need to make it into other apps that use this version of the wallet provider

**Review by adamgall — APPROVED** [review]

**Comment by adamgall** [comment]
@Da-Colon feel free to merge this when you see fit.

Then please open a PR to merge develop into master so we can get the prod website updated with these fixes.

Thanks!

**Pull request merged**